war/src/main/js/add-item.js

```diff
@@ -0,0 +1,414 @@
+    }
+      
+  }
+  
+  function setTabIndex(){
+    $('footer a').attr('tabindex',10);
+    $('#page-head a').attr('tabindex',5);
+    $tabs.find('input, a').attr('tabindex',0);
+    $categories.find('input[type="radio"]').attr('tabindex',0);
+    $('#bottom-sticker').find('button').attr('tabindex',0);  
+    $categories.find('input[type="text"]').attr('tabindex',1);
+    $categories.find('a').attr('tabindex',1);
+  }
+  
+  function watchScroll(e,elem,b,c){
+    var $window = $(window);
+    var toolbarFromTop = $tabs.offset().top;
+    var $jenkTools = $('#breadcrumbBar')
+    var winScoll = $window.scrollTop();
+    var jenkToolOffset = $jenkTools.height() + $jenkTools.offset().top + 15;
+ 
+    $tabs.find('.active').removeClass('active');
+    $.each(data,function(i,cat){
+      var domId = '#'+cat.id;
+      var $cat = $(domId);
+      var catHeight = ($cat.length > 0)?
+          $cat.offset().top + $cat.outerHeight() - (jenkToolOffset + 100):
+            0;
+      if(winScoll < catHeight){
+        var $thisTab = $tabs.find(['[href="',cleanHref(domId),'"]'].join(''));
+        resetActiveTab($thisTab);
+        return false;
+      }
+    });
+    
+    if(winScoll > $('#page-head').height() - 5 ){  
+      $tabs.width($tabs.width()).css({
+        'position':'fixed',
+        'top':($jenkTools.height() - 5 )+'px'});
+      $categories.css({'margin-top':$tabs.outerHeight()+'px'});
+      
+    }
+    else{
+      $tabs.add($categories).removeAttr('style');
+    }
+  }
+
+  
+  function drawTab(i,elem){
+    if(!elem) elem = i;
+    var $tab = $(['<li><a class="tab" href="#',cleanHref(elem.id),'">',elem.display,'</a></li>'].join(''))
+      .click(function(e){
+        //e.preventDefault(e);
+        var $this = $(this).children('a');
+        
+        var tab = $this.attr('href');
+        var scrollTop = $(cleanHref(tab,true)).offset().top - ($newView.children('.jenkins-config-widgets').height() + 15);
+        
+        setTimeout(function(){resetActiveTab($this)},510);
+        
+        $('html,body').animate({
+          scrollTop: scrollTop
+        }, 500);
+      });
+    return $tab;
+  }
+  
+  function drawTabs(data){
+    $('body').addClass('add-item');
+    setTimeout(function(){$('body').addClass('hide-side');},200);
+    $('#main-panel').addClass('container');
+    var $navBox = $('<nav class="navbar navbar-default navbar-static form-config tabBarFrame"/>');
+    var $nav = $('<ul class="nav navbar-nav tabBar config-section-activators" />');
+    
+    $.each(data,function(i,elem){
+      if(!checkCatCount(elem)) return;
+      // little bit hacky here... need to keep track if I have tabs to show, so if there is just 1, I can hide it later....
+      else if (elem.minToShow !== 0) sectionsToShow.push(elem.id);
+      
+      $nav.append(drawTab(elem));
+    });
+    $(window).on('scroll',watchScroll);
+    $navBox.append($nav);
+    $tabs.prepend($navBox);
+    
+    cleanLayout();
+  }
```

Comment:
I am seeing lots of code here around tabbing, but I suspect no effort was made to see how this fits in with the code already in `war/src/main/js/widgets`?